### PR TITLE
Add tests for multi-device and refactor encryption service config

### PR DESCRIPTION
**Pull request by cammellos**
No functional changes,
refactor config so that it can be passed on as an argument and add a test for multi-device.

### CODE REVIEW

File: services/shhext/chat/encryption.go [hunk 1]
```diff
@@ -19,40 +19,51 @@ import (
...
+	log         log.Logger
+	persistence PersistenceService
+	config      EncryptionServiceConfig
+	mutex       sync.Mutex
+}
+
+type EncryptionServiceConfig struct {
+	InstallationID string
+	// Max number of installations we keep synchronized.
+	MaxInstallations uint
```
Comment (dshulyak) [inline services/shhext/chat/encryption.go]: why every other type is regular int? it seems all of them can be only positive. i would keep int for max installation too

File: services/shhext/chat/encryption_multi_device_test.go [hunk 2]
```diff
@@ -12,110 +14,93 @@ func TestEncryptionServiceMultiDeviceSuite(t *testing.T) {
 	suite.Run(t, new(EncryptionServiceMultiDeviceSuite))
 }
 
+type serviceAndKey struct {
+	e []*EncryptionService
```
Comment (dshulyak) [inline services/shhext/chat/encryption_multi_device_test.go]: i love to use this shortcuts myself, but hard to understand what is e when you read code for first times

File: services/shhext/chat/encryption_multi_device_test.go [hunk 3]
```diff
@@ -12,110 +14,93 @@ func TestEncryptionServiceMultiDeviceSuite(t *testing.T) {
...
-	s.alice2 = NewEncryptionService(alicePersistence2, "alice2")
-	s.bob2 = NewEncryptionService(bobPersistence2, "bob2")
+	}
+
+}
 
-	s.alice3 = NewEncryptionService(alicePersistence3, "alice3")
+func (s *EncryptionServiceMultiDeviceSuite) SetupTest() {
+	s.services = make(map[string]*serviceAndKey)
+	setupUser("alice", s, 4)
```
Comment (dshulyak) [inline services/shhext/chat/encryption_multi_device_test.go]: i would use constants for alice and bob. typos..

**Review by dshulyak — APPROVED** [review]
added several styling suggestions

**Comment by cammellos** [comment]
@dmitryn thanks for the feedback, made the changes, if you'd like to have a look

**Comment by cammellos** [comment]
sorry @dmitryn , I meant @dshulyak :), last time I promise

File: services/shhext/chat/encryption_multi_device_test.go [hunk 4]
```diff
@@ -1,121 +1,111 @@
...
-	}
+		os.Remove(dbPath)
 
-	bobPersistence2, err := NewSQLLitePersistence(bobDBPath2, bobDBKey2)
-	if err != nil {
-		panic(err)
-	}
+		persistence, err := NewSQLLitePersistence(dbPath, "key")
+		if err != nil {
+			panic(err)
```
Comment (adambabik) [inline services/shhext/chat/encryption_multi_device_test.go]: I would avoid `panic` in tests. You can just return `error` and assert it in a setup method.

File: services/shhext/chat/encryption_multi_device_test.go [hunk 5]
```diff
@@ -152,25 +142,107 @@ func (s *EncryptionServiceMultiDeviceSuite) TestProcessPublicBundleOutOfOrder()
...
+
+	// Check bundle is ok
+	// No installation is enabled
+	aliceBundle, err := alice1.CreateBundle(aliceKey)
+	s.Require().NoError(err)
+
+	// Check all installations are correctly working, and that the oldest device is not there
+	preKeys := aliceBundle.GetSignedPreKeys()
+	s.Require().Equal(3, len(preKeys))
+	s.Require().NotNil(preKeys["alice1"])
```
Comment (adambabik) [inline services/shhext/chat/encryption_multi_device_test.go:210]: What about `s.Require().NotNil(preKeys["alice2"])`?

Comment (cammellos) [reply services/shhext/chat/encryption_multi_device_test.go:210]: `alice2` should not be there, we only ever propagate `maxInstallations` (in this case is set to 3):
`alice1` is the current device, so it is always included
`alice2` is the oldest device, so it is rotated out, while `alice3`/`alice4` are kept as newer
I could add  `s.Require().Nil(preKeys["alice2"])` to make it more explicit (we count the devices so we know `alice2` is not there)

Comment (adambabik) [reply services/shhext/chat/encryption_multi_device_test.go:210]: I missed that. I think that would be helpful with a comment why `alice2` is not there 👍

Comment (cammellos) [reply services/shhext/chat/encryption_multi_device_test.go:210]: yes, makes sense, already added!

**Review by adambabik — APPROVED** [review]

**Review by pedropombeiro — APPROVED** [review]

**Pull request merged**